util/helm/cmd.go

```diff
@@ -132,40 +220,125 @@ func writeToTmp(data []byte) (string, io.Closer, error) {
 	}), nil
 }
 
-func (c *Cmd) Fetch(repo, chartName, version, destination string, creds Creds) (string, error) {
-	args := []string{c.pullCommand, "--destination", destination}
+func (c *Cmd) Fetch(repoNamespace, repo, chartName, version, destination string, creds Creds) (string, error) {
+	output := ""
+	var err error
+	if c.ociSupported {
+		args := []string{"chart", "pull"}
+		repoUrl := fmt.Sprintf(repo + "/" + repoNamespace + "/" + chartName + ":" + version)
+		log.Infof("Helm OCI-based repository is %s", repoUrl)
 
-	if version != "" {
-		args = append(args, "--version", version)
-	}
-	if creds.Username != "" {
-		args = append(args, "--username", creds.Username)
-	}
-	if creds.Password != "" {
-		args = append(args, "--password", creds.Password)
-	}
-	if creds.CAPath != "" {
-		args = append(args, "--ca-file", creds.CAPath)
-	}
-	if len(creds.CertData) > 0 {
-		filePath, closer, err := writeToTmp(creds.CertData)
+		args = append(args, repoUrl)
+
+		if creds.CAPath != "" {
+			args = append(args, "--ca-file", creds.CAPath)
+		}
+		if len(creds.CertData) > 0 {
+			filePath, closer, err := writeToTmp(creds.CertData)
+			if err != nil {
+				return "", err
+			}
+			defer io.Close(closer)
+			args = append(args, "--cert-file", filePath)
+		}
+		if len(creds.KeyData) > 0 {
+			filePath, closer, err := writeToTmp(creds.KeyData)
+			if err != nil {
+				return "", err
+			}
+			defer io.Close(closer)
+			args = append(args, "--key-file", filePath)
+		}
+
+		_, err := c.run(args...)
 		if err != nil {
 			return "", err
 		}
-		defer io.Close(closer)
-		args = append(args, "--cert-file", filePath)
-	}
-	if len(creds.KeyData) > 0 {
-		filePath, closer, err := writeToTmp(creds.KeyData)
+
+		// helm chart export
+		args = []string{"chart", "export", repoUrl, "--destination", destination}
+
+		if creds.CAPath != "" {
+			args = append(args, "--ca-file", creds.CAPath)
+		}
+		if len(creds.CertData) > 0 {
+			filePath, closer, err := writeToTmp(creds.CertData)
+			if err != nil {
+				return "", err
+			}
+			defer io.Close(closer)
+			args = append(args, "--cert-file", filePath)
+		}
+		if len(creds.KeyData) > 0 {
+			filePath, closer, err := writeToTmp(creds.KeyData)
+			if err != nil {
+				return "", err
+			}
+			defer io.Close(closer)
+			args = append(args, "--key-file", filePath)
+		}
+
+		output, err = c.run(args...)
+		if err != nil {
+			return "", err
+		}
+
+		// tar helm chart
+		cmd := exec.Command("tar", "-zcvf", repoNamespace+"-"+chartName+"-"+version+".tgz", chartName)
```

Comment:
Why is running `tar` necessary here?

Reply:
The helm type repository will fetch a tar package, but helm-oci type repository is exported as a folder,  I want to make them both be a tar package,  so the codes below can keep handle them the same way.

Reply:
> Why is running `tar` necessary here?

The helm type repository will fetch a tar package, but helm-oci type repository is exported as a folder, I want to make them both be a tar package, so the codes below can keep handle them the same way.